lib/timescaledb/stats/hypertables.rb

```diff
@@ -0,0 +1,86 @@
+require 'bytesize'
+
+require_relative './chunks'
+
+module Timescaledb
+  class Stats
+    class Hypertables
+      # @param [Timescaledb:Connection] connection The PG connection.
+      # @param [Array<String>] hypertables The list of hypertable names.
+      def initialize(connection, hypertables = [])
+        @connection = connection
+        @hypertables = hypertables
+      end
+
+      delegate :query, :query_first, :query_count, to: :@connection
+
+      # @return [Hash] The hypertables stats
+      def to_h
+        {
+          count: @hypertables.count,
+          uncompressed_count: uncompressed_count,
+          approximate_row_count: approximate_row_count,
+          chunks: Timescaledb::Stats::Chunks.new(@connection, @hypertables).to_h,
+          size: size
+        }
+      end
+
+      private
+
+      def uncompressed_count
+        @hypertables.count do |hypertable|
+          query("SELECT * from hypertable_compression_stats('#{hypertable}')").empty?
```

Comment:
Can we make this query more efficient? Probably we can do the counting using the chunks view.

If we would have the ActiveRecord support, it would be something like:

```ruby
Chunk.where(hypertable_name: @hypertales).uncompressed.count
```

Probably a single query will be faster.

Reply:
I tried to implement everything as it was already implemented with Active Record, for this part I used 👇 , I can memoize the value as you were doing or I can do what you suggested if you feel is better, let me know and I will change it 🚀 

https://github.com/jonatas/timescaledb/blob/489a9abd6288aeff8edf04370188d946ff4a8f82/lib/timescaledb/stats_report.rb#L11 and https://github.com/jonatas/timescaledb/blob/489a9abd6288aeff8edf04370188d946ff4a8f82/lib/timescaledb/hypertable.rb#L29-L32

Reply:
As we discussed, another change will introduce this improvement 💪